include/modules/hyprland/workspaces.hpp

```diff
@@ -43,19 +46,27 @@ class Workspaces : public AModule, public EventHandler {
   void update() override;
   void init();
 
+  auto all_outputs() const -> bool { return all_outputs_; }
+  auto show_special() const -> bool { return show_special_; }
+
+  auto get_bar_output() const -> std::string { return bar_.output->name; }
+
  private:
   void onEvent(const std::string&) override;
   void sort_workspaces();
-  void create_workspace(int id);
-  void remove_workspace(int id);
+  void create_workspace(const Json::Value& value);
```

Comment:
I would prefer leave Dto struct. Json value hints nothing about what's inside it, but Dto does.
And the reason why id was here -- is because it's clearly to see what's going on inside `onEvent` methond and `update` method. Push Json value through methods -- it' a mess. You can compare current solution with sway solution.

Reply:
this and a combined destroy/remove queue is something that should be done, but maybe in the future, since it'll require a rewrite of most of the code.